azerothcore/azerothcore-wotlk
# feat(Core/Database): port TrinityCore database API

## Changes Proposed:
- It's about time, hah
- Port all changes TrinityCore database

## Issues Addressed:
- Closes #4669
- Part #3739 (code only, start support after add boost suport)
- Maybe #4005
- Maybe #3679

## Tests Performed:
- Char (Loading/create/delete)
- Hunter pets (loading, combat, stable master, rename)

## How to Test the Changes:
- Check char create/rename/customize/delete/loading and other
- Check all pets functions (rename/add) and other
- Check character save system

## Known Issues and TODO List:
- [ ] https://github.com/ElunaLuaEngine/Eluna/pull/358
- [ ] https://github.com/azerothcore/mod-transmog/pull/45
- [x] https://github.com/azerothcore/azerothcore-wotlk/pull/5671
- [x] #5950
- [x] Fix `clang-7`
- [ ] change wh
- [x] `.char rename`
- [x] `.char customize`

## Target Branch(es):
 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
